extra/markdown-extra/LeagueMarkdownFactory.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Extra\Markdown;
+
+use League\CommonMark\CommonMarkConverter;
+use League\CommonMark\Extension\ExtensionInterface;
+
+final class LeagueMarkdownFactory
+{
+    private $extensions;
+    private $converter;
+
+    /**
+     * @param ExtensionInterface[] $extensions
+     */
+    public function __construct(iterable $extensions, CommonMarkConverter $converter = null)
+    {
+        $this->extensions = $extensions;
+        $this->converter = $converter ?: new CommonMarkConverter();
+    }
+
+    public function __invoke(): LeagueMarkdown
+    {
+        foreach ($this->extensions as $extension) {
+            $this->converter->getEnvironment()->addExtension($extension);
+        }
+
+        return new LeagueMarkdown($this->converter);
```

Comment:
Why making this class responsible for instantiating the LeagueMarkdown ? The work that the DI component cannot do by itself is instantiating the CommonMarkConverter with extensions.